test/cppsuite/test_harness/workload_tracking.h

```diff
@@ -30,39 +30,77 @@
 #define WORKLOAD_TRACKING_H
 
 /*
- * Default schema for tracking table key_format : Collection name / Key value_format : Operation
- * type / Value / Timestamp
+ * Default schema for tracking operations on collections (key_format: Collection name / Key /
+ * Timestamp, value_format: Operation type / Value)
  */
-#define DEFAULT_TRACKING_KEY_FORMAT WT_UNCHECKED_STRING(Si)
-#define DEFAULT_TRACKING_VALUE_FORMAT WT_UNCHECKED_STRING(iSi)
-#define DEFAULT_TRACKING_TABLE_SCHEMA \
-    "key_format=" DEFAULT_TRACKING_KEY_FORMAT ",value_format=" DEFAULT_TRACKING_VALUE_FORMAT
+#define OPERATION_TRACKING_KEY_FORMAT WT_UNCHECKED_STRING(Sii)
+#define OPERATION_TRACKING_VALUE_FORMAT WT_UNCHECKED_STRING(iS)
+#define OPERATION_TRACKING_TABLE_CONFIG \
+    "key_format=" OPERATION_TRACKING_KEY_FORMAT ",value_format=" OPERATION_TRACKING_VALUE_FORMAT
+
+/*
+ * Default schema for tracking schema operations on collections (key_format: Collection name /
+ * Timestamp, value_format: Operation type)
+ */
+#define SCHEMA_TRACKING_KEY_FORMAT WT_UNCHECKED_STRING(Si)
+#define SCHEMA_TRACKING_VALUE_FORMAT WT_UNCHECKED_STRING(i)
+#define SCHEMA_TRACKING_TABLE_CONFIG \
+    "key_format=" SCHEMA_TRACKING_KEY_FORMAT ",value_format=" SCHEMA_TRACKING_VALUE_FORMAT
 
 namespace test_harness {
 /* Tracking operations. */
-enum class tracking_operation { CREATE, INSERT };
+enum class tracking_operation { CREATE, DELETE_COLLECTION, DELETE_KEY, INSERT };
 /* Class used to track operations performed on collections */
-class workload_tracking {
+class workload_tracking : public component {
 
     public:
-    workload_tracking(const std::string &collection_name)
-        : _collection_name(collection_name), _timestamp(0U)
+    workload_tracking(const std::string &operation_table_format,
```

Comment:
Rather than passing in the format of the table, we should only be passing in the format of the key value pairs that are being generated in the workload generator. The rest of the schema for the tables will remain the same. The scenario I have in mind is that currently the workload generator writes integer keys and string values, but we will extend it to write string k/v pairs. As such really the tracking component should only know those formats so it can incorporate them. The rest of the schema should be set in by the tracking component.

Reply:
As discussed, it will be resolved in another ticket as this one already has lots of changes and this is getting out of scope.